ios/Classes/PTPluginUtils.m

```diff
@@ -20,26 +58,385 @@ + (NSNumber *)PT_idAsNSNumber:(id)value
+                        }
+                    }
+                }
+            }
+        }
+        
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: There was an error while trying to export annotations. %@", error.localizedDescription);
+    }
+    
+    [docVC.toolManager changeTool:[PTPanTool class]];
+    
+    flutterResult(nil);
+}
+
++ (void)selectAnnotation:(NSString *)annotation resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    NSError* error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if([doc HasDownloader])
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
+
+        NSDictionary *annotationJson = [PTPluginUtils PT_idAsNSDict:[PTPluginUtils PT_JSONStringToId:annotation]];
+        
+        NSString *annotId = [PTPluginUtils PT_idAsNSString:annotationJson[PTAnnotIdKey]];
+        int pageNumber = [[PTPluginUtils PT_idAsNSNumber:annotationJson[PTAnnotPageNumberKey]] intValue];
+        
+        PTAnnot *annot = [PTPluginUtils findAnnotWithUniqueID:annotId onPageNumber:pageNumber documentViewController:docVC];
+        
+        [docVC.toolManager selectAnnotation:annot onPageNumber:pageNumber];
+
+        [docVC.pdfViewCtrl UpdateWithAnnot:annot page_num:pageNumber];
+
+    } error:&error];
+    
+    if(error)
+    {
+        NSLog(@"Error: There was an error while trying to select annotation. %@", error.localizedDescription);
     }
+    
+    flutterResult(nil);
 }
 
++ (void)setFlagForAnnotations:(NSString *)annotationsWithFlags resultToken:(FlutterResult)flutterResult documentViewController:(PTDocumentViewController *)docVC
+{
+    if(docVC.document == Nil)
+    {
+        // something is wrong, no document.
+        NSLog(@"Error: The document view controller has no document.");
+        return;
+    }
+    
+    NSError* error;
+    
+    [docVC.pdfViewCtrl DocLock:YES withBlock:^(PTPDFDoc * _Nullable doc) {
+        if([doc HasDownloader])
+        {
+            // too soon
+            NSLog(@"Error: The document is still being downloaded.");
+            return;
+        }
 
-+ (void)importAnnotationCommand:(NSString *)xfdfCommand documentViewController:(PTDocumentViewController *)docVC
+        NSArray *annotationWithFlagsArray = [PTPluginUtils PT_idAsArray:[PTPluginUtils PT_JSONStringToId:annotationsWithFlags]];
+        
+        for (NSDictionary *currentAnnotationWithFlags in annotationWithFlagsArray)
+        {
+            NSDictionary *currentAnnotation = [PTPluginUtils PT_idAsNSDict:[PTPluginUtils PT_JSONStringToId:currentAnnotationWithFlags[PTAnnotationArgumentKey]]];
```

Comment:
It's a bit confusing you have two vars here, one called `currentAnnotation` and another called `currentAnnot`. Can one of these be renamed to make it more clear how they are different?